src/ui/atoms/textField/TextField.tsx

```diff
@@ -20,43 +20,30 @@ export type TextFieldProps = InputBaseProps & {
    * If true, the TextField will take 100% of its parent's size.
    */
   readonly fullWidth?: boolean
-  /**
-   * If true, the TextField allows multiline text as a text area.
-   */
-  readonly multiline?: boolean
-  /**
-   * If multiline, defines the number of lines of the text area.
-   */
-  readonly numberOfLines?: number
-  /**
-   * If true and multiline, the user can't resize the text area.
-   */
-  readonly disableAreaResize?: boolean
 }
 
 export const TextField: React.FC<TextFieldProps> = ({
   size = 'medium',
   helperText,
   fullWidth = false,
   multiline = false,
-  numberOfLines,
   disableAreaResize = false,
   ...props
 }: TextFieldProps): JSX.Element => {
   const containerClass = classNames(
     `okp4-text-field-main ${multiline && !disableAreaResize ? 'auto' : size}`,
     {
-      'full-width': !multiline && fullWidth
+      'full-width': fullWidth
     }
   )
 
   return (
     <div className={containerClass}>
       <InputBase
+        {...props}
         disableAreaResize={disableAreaResize}
         multiline={multiline}
-        numberOfLines={numberOfLines}
-        {...props}
+        resizeHeightOnly={fullWidth}
```

Comment:
What about if the user set `resizeHeightOnly` to `false` event if you are in `fullWidth` mode?

There is no effect as the value is based upon `fullWidth`'s one .. So it can lead to a misunderstood for the user which is not good...